Makefile

```diff
@@ -0,0 +1,53 @@
+OBJECTS := JSON.o Monster.o Hero.o main.o
+COMPILERFLAGS := -std=c++17 -Wall -Werror -Wextra 
+COMPILER := g++
+
+CPPFILES := JSON.cpp Monster.cpp Hero.cpp main.cpp
+
+VALGRINDFLAGS:= --leak-check=full --error-exitcode=1
+VALGRINDPARAMETER:=  ./a.out scenario1.json
+
+build: $(OBJECTS)
+	$(COMPILER) $(COMPILERFLAGS) -o a.out $(OBJECTS)
+
+JSON.o: JSON.cpp JSON.h
+	$(COMPILER) $(COMPILERFLAGS) -c JSON.cpp
+
+Monster.o: Monster.cpp Monster.h JSON.h
+	$(COMPILER) $(COMPILERFLAGS) -c Monster.cpp
+
+Hero.o: Hero.cpp Hero.h Monster.h JSON.h
+	$(COMPILER) $(COMPILERFLAGS) -c Hero.cpp
+
+main.o: main.cpp Monster.h Hero.h
```

Comment:
Es a Hero nem hasznalja a JSON.h-t?

Reply:
Itt lehet, hogy valamit benéztem, de nem értem a hibát.

Reply:
Az lenne, hogy a Hero.o előállításához nincs szükség a JSON headerre, mert a Monster már includeolja?

Reply:
Igy van. az is bekerul oda kozvetetten. Ezert ha az valtozik, ezt is ujra kell forditani.

Reply:
Akkor elnézést a kérdésemért, lehet hogy én értem félre, de a "Hero nem használja a JSON.h-t" nem lehetett elírás? Nem igazán esett le nekünk az alapján a probléma, de ha "monster" lenne a mondatban akkor minden világossá válna ezek alapján.

Reply:
Hat kerdes volt ugye, nem kijelentes :-) Es valoszinuleg a Monster es a Hero is hasznalja a parse fuggvenyeben.

De most mar megneztem akkor a kodotokat. Es tok okosan a JSON.h include-ot cska a cpp-be tettetek bele, igy ide akkor valoban nem kell. A kerdesre ha ezt megirtatok  volna, akkor case closed :-)

Viszont most latom, hogy "Az lenne, hogy a Hero.o előállításához nincs szükség a JSON headerre, mert a Monster már includeolja?" <-- ezt teljesen felreertettem. Valami masra, nem tudom, mire, most elolvasva igazabol sehogy sem ertem. :-D A lenyeg, hogy minden precompile stepben ugye az include-ok vegrehajtodnak, es azok egy csomo header file-t copypaster-elnek. Ha azokban van valtozas, akkor is ujra kell forditani a cpp-t. Akkor is, ha nem kozvetlenul rantottuk be, hanem monjduk egy Hero.h-t rantottunk be, ami berantott egy Monster.h-t, ami berantott egy JSON.h-t.  Ha kint lett volna a JSON.h include a headerben, akkor itt is kellett volna. igy nem kell, viszem is be.